File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

function solve(root, curr, sum) {
    if(root == null) return;
    curr = curr * 10 + root.val;
    if(root.left == null && root.right == null) {
        sum[0] += curr;
    }

    solve(root.left, curr, sum);
    solve(root.right, curr, sum);
};

var sumNumbers = function(root) {
    if(root == null) return 0;
    let sum = Array(1).fill(0);
    solve(root, 0, sum);
    return sum;
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def solve(self, root, curr, sum):
        if root == None:
            return
        curr = curr * 10 + root.val
        if root.left == None and root.right == None:
            sum[0] += curr
        
        self.solve(root.left, curr, sum)
        self.solve(root.right, curr, sum)

    def sumNumbers(self, root):
        if root == None:
            return 0
        sum = [0] * 1
        self.solve(root, 0, sum)
        return sum[0]
        